src/senaite/core/catalog/indexer/client.py

```diff
@@ -45,4 +45,6 @@ def client_searchable_text(instance):
     tokens = filter(None, set(tokens))
 
     # return a single unicode string with all the concatenated tokens
-    return u" ".join(map(api.safe_unicode, tokens))
+    return u" ".join(map(lambda x: api.safe_unicode(str(x)), tokens))
```

Comment:
Hi @francoisler-lab,
This line will fail if a token comes in already as unicode, e.g.
```python
In [1]: str(u"März")
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-1-b1687a0d3492> in <module>()
----> 1 str(u"März")

UnicodeEncodeError: 'ascii' codec can't encode character u'\xe4' in position 1: ordinal not in range(128)
```
Better do a `for token in tokens` loop and check each token if it is a string type or not.

Hint:  You can use `api.is_string` from https://github.com/senaite/senaite.core/blob/2.x/src/bika/lims/api/__init__.py#L2028